Sources/NIO/BaseStreamSocketChannel.swift

```diff
@@ -72,6 +72,20 @@ class BaseStreamSocketChannel<Socket: SocketProtocol>: BaseSocketChannel<Socket>
         }
     }
 
+    // Hook for customizable socket shutdown processing for subclasses, e.g. PipeChannel
+    func shutdownSocket(mode: CloseMode) throws {
+        switch mode {
+            case .output:
```

Comment:
Nit: NIO indentation here would be:

```
switch mode {
case .output:
```

Reply:
Ok (but ouch for my personal taste :-) - fixed both and pushed...